signer/src/stacks/api.rs

```diff
@@ -1476,10 +1327,45 @@ where
                     );
                     break;
                 }
+                // A 404 could also mean there is a bitcoin block with no stacks
+                // blocks anchored.
+                // TODO: can there be other reasons for a 404?
+                if let Error::StacksNodeResponse(ref reqwest_error) = error
+                    && reqwest_error.status() == Some(StatusCode::NOT_FOUND)
+                {
+                    continue;
+                }
                 return Err(error);
             }
         };
-        tenure = tenure_headers;
+        let newest_block_in_older_tenure_height = tenure_headers.end_height();
+        let newest_block_in_older_tenure = tenure_headers
+            .headers
+            .iter()
+            .find(|header| header.block_height == newest_block_in_older_tenure_height);
+        let oldest_block_in_newer_tenure_height = tenure.start_height();
+        let oldest_block_in_newer_tenure = tenure
+            .headers
+            .iter()
+            .find(|header| header.block_height == oldest_block_in_newer_tenure_height);
```

Comment:
This is okay, but we can skip the double iteration by changing the `TenureBlockHeaders::start_height` and `TenureBlockHeaders::end_height` functions into a `start_header` and `end_header` functions that return the respective headers. Then we can just compare the hash here.

Reply:
[f932fa6](https://github.com/stacks-sbtc/sbtc/pull/1876/commits/f932fa6bb045bd0ee0a3d4e41539025442b22fe5)